src/test/java/com/icthh/xm/tmf/ms/communication/messaging/template/TwilioMessageTemplateServiceUnitTest.java

```diff
@@ -0,0 +1,113 @@
+import com.icthh.xm.tmf.ms.communication.web.api.model.Sender;
+import freemarker.cache.StringTemplateLoader;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.ArrayList;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static org.rnorth.visibleassertions.VisibleAssertions.assertThrows;
+
+public class TwilioMessageTemplateServiceUnitTest extends AbstractSpringBootTest {
+
+    private static final String TENANT = "TEST";
+    private static final String TEMPLATE_NAME = "templateName";
+    private static final Locale LOCALE = new Locale("uk","UA");
+
+    private static final String PATH_PATTERN = "/config/tenants/{tenantKey}/communication/msisdn/{langKey}/{templateName}.txt";
+    private static final String CONFIG_PATH = "/config/tenants/TEST/communication/msisdn/uk/templateName.txt";
+    private static final String CONFIG = "Hello, ${user.firstName + ' ' + user.lastName}! This is your code: ${code}";
+
+    @Autowired
+    private freemarker.template.Configuration freeMarkerConfiguration;
+
+    @Autowired
+    private StringTemplateLoader templateLoader;
+
+    @Autowired
+    private MultiTenantLangStringTemplateLoaderService templateLoaderService;
+
+    @Mock
+    private ApplicationProperties applicationProperties;
+
+    @Mock
+    private MessageTemplateConfigurationService messageTemplateConfigurationService;
+
+    private TwilioMessageTemplateService twilioMessageTemplateService;
+
+    @Before
+    public void setUp() {
+        when(applicationProperties.getMsisdnPathPattern()).thenReturn(PATH_PATTERN);
+        when(messageTemplateConfigurationService.getMsisdnTemplateContent(CONFIG_PATH)).thenReturn(CONFIG);
+
+        twilioMessageTemplateService = new TwilioMessageTemplateService(freeMarkerConfiguration, templateLoader,
+            templateLoaderService, applicationProperties, messageTemplateConfigurationService);
+    }
+
+    @Test
+    public void getMessageContent_emptyData_shouldThrowException() {
+        Map<String, Object> model = Map.of(
+            "code", "123456789",
+            "user", Map.of("firstName", "John", "lastName", "Smith")
+        );
+
+        assertThrows("Language key, template name and tenant must be not blank",
+            IllegalStateException.class, () -> {
+            twilioMessageTemplateService.getMessageContent("", TEMPLATE_NAME, LOCALE, model);
+        });
+
+        assertThrows("Language key, template name and tenant must be not blank",
+            IllegalStateException.class, () -> {
+                twilioMessageTemplateService.getMessageContent(TENANT, null, LOCALE, model);
+            });
+    }
+
+    @Test
+    public void getMessageContent_invalidTemplate_shouldThrowException() throws JsonProcessingException {
```

Comment:
body of test and test name does not much

Reply:
fixed